### PR TITLE
Remove XML files for min-size package

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Remove XML files for min-size package.
The min-size package is supposed to:
1. avoid R2R images for PowerShell and all other library assemblies.
2. not include the Graphical Host and the WPF/WinForm stuff it pulls in.	
3. remove symbol files (.pdb) and XML document files (.xml)

However, not sure since when, the min-size package for Windows started to have XML files. This PR removes the XML files for Windows min-size package.

Test run build: https://dev.azure.com/mscodehub/PowerShellCore/_build/results?buildId=299968&view=results

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -1602,14 +1602,14 @@ function Show-PSPesterError
         $description = $testFailure.description
         $name = $testFailure.name
         $message = $testFailure.failure.message
-        $StackTrace = $testFailure.failure."stack-trace"
+        $stack_trace = $testFailure.failure."stack-trace"
```
Comment (daxian-dbw) [inline build.psm1:1605]: This is to fix a warning from PSSA -- `$StackTrace` is a built-in variable, so should use a different name.

File: build.psm1 [hunk 2]
```diff
@@ -2846,7 +2846,6 @@ assembly
                 # only create an assembly group if we have tests
                 if ( $tCases.count -eq 0 -and ! $includeEmpty ) { continue }
                 $tGroup = $tCases | Group-Object result
-                $total = $tCases.Count
```
Comment (daxian-dbw) [inline build.psm1:2849]: Also for fixing a PSSA warning -- not used variable.

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -4600,14 +4599,14 @@ function Invoke-AzDevOpsLinuxPackageBuild {
             # Remove symbol files.
             Remove-Item "${buildFolder}\*.pdb" -Force
         } elseif ($BuildType -eq 'rpm') {
-            ## Build 'min-size'
+            ## Build for Mariner
```
Comment (daxian-dbw) [inline tools/packaging/packaging.psm1:4602]: This block is for building mariner package, not min-size.

**Comment by iSazonov** [comment]
Perhaps it makes sense to remove resource dll-s too - we have many subfolders with them.

**Comment by daxian-dbw** [comment]
I think it's up to the user of the min-size package to further prune the files based on their scenario. For example, you may want to leave the `ru` folder around and delete the others :).

**Comment by daxian-dbw** [comment]
@adityapatwardhan The test build https://dev.azure.com/mscodehub/PowerShellCore/_build/results?buildId=299968&view=results finished successfully. I have check the produced `win-x64-gc.zip` package, and the XML files are gone.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@daxian-dbw it may be appropriate to have min-size be English only for now and we can add back other languages based on user feedback.  Since PS7 itself isn't localized right now, I suspect most users are ok with just English strings.

**Comment by daxian-dbw** [comment]
> Since PS7 itself isn't localized right now, I suspect most users are ok with just English strings.

@SteveL-MSFT Then, shall we remove the resource dlls from all of our packages?

For min-size package itself, it's only for the Guest Config team to consume as of today. Let me check with them and see if they are OK with removing all resource dlls (en-us resource are embedded in the real assemblies).

I'm not going to change this PR since it will be backported to 7.3 and 7.2. If the Guest Config team agrees, I can submit a separate PR to remove the resource dlls.

**Pull request merged**